ui/app/controllers/evaluations/index.js

```diff
@@ -1,12 +1,83 @@
 import Controller from '@ember/controller';
 import { action } from '@ember/object';
 import { tracked } from '@glimmer/tracking';
+import { schedule } from '@ember/runloop';
 import { inject as service } from '@ember/service';
+import { useMachine } from 'ember-statecharts';
+import { use } from 'ember-usable';
+import evaluationsMachine from '../../machines/evaluations';
 
 export default class EvaluationsController extends Controller {
+  @service store;
   @service userSettings;
 
-  queryParams = ['nextToken', 'pageSize', 'status'];
+  @tracked width = null;
+  @tracked height = null;
+
+  @use
+  statechart = useMachine(evaluationsMachine).withConfig({
+    services: {
+      loadEvaluation: this.loadEvaluation,
+    },
+    actions: {
+      updateEvaluationQueryParameter: this.updateEvaluationQueryParameter,
+      removeCurrentEvaluationQueryParameter:
+        this.removeCurrentEvaluationQueryParameter,
+    },
+    guards: {
+      sidebarIsOpen: this._sidebarIsOpen,
+    },
+  });
+
+  queryParams = ['nextToken', 'currentEval', 'pageSize', 'status'];
+  @tracked currentEval = null;
+
+  @action
+  _sidebarIsOpen() {
+    return !!this.currentEval;
+  }
+
+  @action
+  async loadEvaluation(context, { evaluation }) {
+    let evaluationId;
+    if (evaluation?.id) {
+      evaluationId = evaluation.id;
+    } else {
+      evaluationId = this.currentEval;
+    }
+
+    return this.store.findRecord('evaluation', evaluationId, {
+      reload: true,
+      adapterOptions: { related: true },
+    });
+  }
+
+  @action
+  async handleEvaluationClick(evaluation, e) {
+    if (
+      e instanceof MouseEvent ||
+      (e instanceof KeyboardEvent && (e.code === 'Enter' || e.code === 'Space'))
```

Comment:
we might want to consider using [https://adopted-ember-addons.github.io/ember-keyboard/](https://adopted-ember-addons.github.io/ember-keyboard/) when we plan to use more keyboard navigation in the application